server/server.php

```diff
@@ -1,105 +1,213 @@
 <?php
 
-define("SOCKET_PORT", 8080);
-
-use Ratchet\Server\IoServer;
-use Ratchet\Http\HttpServer;
-use Ratchet\WebSocket\WsServer;
-use Ratchet\MessageComponentInterface;
-use Ratchet\WebSocket\WsServerInterface;
-use Ratchet\ConnectionInterface;
+use \WebSocket\Throwable;
+use \WebSocket\Exception\ExceptionInterface;
+use \WebSocket\Message\{
+    Binary,
+    Close,
+    Ping,
+    Pong,
+    Text,
+};
+use \WebSocket\Middleware\{
+    CloseHandler,
+    CompressionExtension,
+    PingResponder,
+};
+use \WebSocket\Middleware\CompressionExtension\DeflateCompressor;
+use \WebSocket\Test\EchoLog;
+
 require dirname(dirname( __FILE__ )) . '/vendor/autoload.php';
 
-class Socket implements MessageComponentInterface, WsServerInterface {
+$options = [
+    "port" => 8080,
+    "ssl" => true,
+    "timeout" => "60",
+    "framesize" => "4096",
+    "connections" => 30,
+    "debug" => false,
+    "deflate" => false
+];
+
+class SolidPubSub {
+    private $options;
+    private $clients;
+    private $subscriptions;
 
-    public function __construct()
-    {
-        $this->clients = new \SplObjectStorage;
-		$this->subscriptions = array();
-		$this->subprotocols = array("solid-0.1");
+    public function __construct($options) {
+        $this->options = $options;
+        $this->clients = [];
+        $this->subscriptions = [];
+        $this->server = $this->createServer();
+    }
+    
+    public function onHandshake(
+        \WebSocket\Server $server,
+        \WebSocket\Connection $connection,
+        $request,
+        $response
+    ) {
+      // Store the new connection in $this->clients
+        $this->clients[$connection->getRemoteName()] = $connection;
+        echo "> [{$connection->getRemoteName()}] Client connected {$request->getUri()}\n";
```

Comment:
For the sake of readability, it might be worth using `printf` instead of `echo`. Consider:

```php
vprintf("> [%s] Client connected %s\n", [
    $connection->getRemoteName(),
    $request->getUri(),
]);